src/Apps/Order/Routes/Shipping/index.tsx

```diff
@@ -838,8 +850,11 @@ export const ShippingRoute: FC<ShippingProps> = props => {
                   ) => {
                     setAddressNeedsVerification(false)
                     setAddressVerifiedBy(verifiedBy)
-                    setAddress({ ...address, ...chosenAddress })
-                    saveAndContinue && finalizeFulfillment()
+                    setAddress(address => {
+                      return { ...address, ...chosenAddress }
+                    })
+                    setReadyToSaveVerifiedAddress(true)
+                    // saveAndContinue && finalizeFulfillment()
```

Comment:
by the time this finalizeFulfillment has executed it has been redefined due to the changes above, but the stale function value persists in the closure of the function passed as the `onChosenAddress` prop.

Reply:
Nice! So we should probably not rely on any data that's not passed in from the function arguments. Guess using a function in the set function above is more reliable in the same way.